src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1257,6 +1238,40 @@ void PDP::set_initial_announcement_interval()
     set_next_announcement_interval();
 }
 
+void PDP::set_external_participant_properties_(ParticipantProxyData* pproxy)
+{
+    // For each property add it if it should be sent (it is propagated)
+    for (auto const& property : mp_RTPSParticipant->getAttributes().properties.properties())
+    {
+        if(property.propagate())
+        {
+            pproxy->m_properties.push_back(property.name(), property.value());
+        }
+    }
+
+    // Set participant type property
+    std::stringstream participant_type;
+    participant_type << mp_RTPSParticipant->getAttributes().builtin.discovery_config.discoveryProtocol;
+    auto ptype = participant_type.str();
+    pproxy->m_properties.push_back(fastdds::dds::parameter_property_participant_type, ptype);
+
+    /* Add physical properties if present */
+    std::vector<std::string> physical_property_names = {
+        fastdds::dds::parameter_policy_physical_data_host,
+        fastdds::dds::parameter_policy_physical_data_user,
+        fastdds::dds::parameter_policy_physical_data_process
+    };
+    for (auto physical_property_name : physical_property_names)
```

Comment:
Two independent suggestions:

1. Enclose this block in a `#ifdef FASTDDS_STATISTICS`
2. Remove this block, and add these properties with `propagated` set to `true` so they are added in the loop above like any other property to be propagated.